cmd/doctor.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/urfave/cli"
+)
+
+// CmdDoctor represents the available doctor sub-command.
+var CmdDoctor = cli.Command{
+	Name:        "doctor",
+	Usage:       "Diagnose the problems",
+	Description: "A command to diagnose the problems of current gitea instance according the given configuration.",
+	Action:      runDoctor,
+}
+
+type check struct {
+	title string
+	f     func(ctx *cli.Context) ([]string, error)
+}
+
+var checklist = []check{
+	{
+		title: "Check if openssh authorized_keys file id correct",
+		f:     runDoctorLocationMoved,
+	},
+}
+
+func runDoctor(ctx *cli.Context) error {
+	err := initDB()
+	fmt.Println("Using app.ini at ", setting.CustomConf)
+	if err != nil {
+		fmt.Println(err)
+		fmt.Println("Check if you are using the right config file. You can use a --config directive to specify one.")
+		return nil
+	}
+
+	for i, check := range checklist {
+		fmt.Println("[", i+1, "]", check.title)
+		fmt.Println()
+		if messages, err := check.f(ctx); err != nil {
+			fmt.Println("Error:", err)
+		} else if len(messages) > 0 {
+			for _, message := range messages {
+				fmt.Println("-", message)
+			}
+		} else {
+			fmt.Println("OK.")
+		}
+		fmt.Println()
+	}
+	return nil
+}
+
+func exePath() (string, error) {
+	file, err := exec.LookPath(os.Args[0])
+	if err != nil {
+		return "", err
+	}
+	return filepath.Abs(file)
+}
+
+func runDoctorLocationMoved(ctx *cli.Context) ([]string, error) {
+	if setting.SSH.StartBuiltinServer {
```

Comment:
This should also be disabled if SSH_CREATE_AUTHORIZED_KEYS_FILE=false 
```suggestion
	if setting.SSH.StartBuiltinServer || !setting.SSH.CreateAuthorizedKeysFile {
```
Similar like the test https://github.com/go-gitea/gitea/blob/dce22efbee6a04b8c19348dac831cd88b18ac07c/models/ssh_key.go#L653